comp/otelcol/collector/pipeline.go

```diff
@@ -65,7 +65,7 @@ type collector struct {
 	col  *otlp.Pipeline
```

Comment:
I think I'm confused. How does the type already defined here in `pipeline.go` relate to the collector being defined in `collector/impl/collector.go` in this PR?

* Is the new collector going to replace this one?
* Are they alternative implementations and only one should be in use at a time?
* Are they compatible implementations and both should be usable at once?
* Are they doing different things entirely and just named in a similar manner?